openmc/tallies.py

```diff
@@ -1468,32 +1528,20 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Query the mean and std dev so the tally data is read in from file
+        # if it has not already been read in.
+        self.mean, self.std_dev, other.mean, other.std_dev
```

Comment:
I like this compressed form, but if @paulromano may prefer that this be made PEP8 compliant. Honestly, I think we need a `load_tally_data(...)` and/or `clear_tally_data(...)` methods which do this with optional parameters to specify the "sum," "mean," etc., but understand this is outside the scope of this PR.